src/main/java/com/evoter/candidate/service/CandidateServiceImpl.java

```diff
@@ -24,33 +31,52 @@ public Candidate createCandidate(CreateUpdateCandidateRequestDto requestDto) {
         log.info("Request to create a candidate with payload={}", requestDto);
 
         // validate that candidate details are not null
+        if(requestDto ==null) throw new GeneralException(ResponseCodeAndMessage.INVALID_JSON_REQUEST_DATA_90
+                .responseCode,"request cannot be null");
 
         // check if candidate already exist
-
-        //save
         Candidate candidate = new Candidate();
         candidate.setName(requestDto.getName());
         candidate.setAge(requestDto.getAge());
         candidate.setSex(requestDto.getSex());
         candidate.setPartyId(requestDto.getPartyId());
         candidate.setPollTypeId(requestDto.getPollTypeId());
+        if(candidateExists(candidate)) throw new GeneralException(ResponseCodeAndMessage.ALREADY_EXIST_86.
+                responseCode,"Candidates already exists");
 
+        //save
         return candidateRepository.save(candidate);
     }
 
+    public boolean candidateExists(Candidate candidate) {
+        log.info("Checks if candidate exists in the database{}",candidate);
+        String query = "SELECT c FROM Candidate c WHERE c.name = :name AND c.sex = :sex AND c.partyId = " +
+                ":partyId AND c.age = :age AND c.pollTypeId = :pollTypeId";
+        TypedQuery<Candidate> typedQuery = entityManager.createQuery(query, Candidate.class)
+                .setParameter("name", candidate.getName())
+                .setParameter("sex", candidate.getSex())
+                .setParameter("partyId", candidate.getPartyId())
+                .setParameter("age",candidate.getAge())
+                .setParameter("pollTypeId",candidate.getPollTypeId());
+
+        List<Candidate> candidates = typedQuery.getResultList();
+
+        return !candidates.isEmpty();
+    }
     @Override
     public List<Candidate> getAllCandidates() {
-        return null;
+        return candidateRepository.findAll();
     }
 
     @Override
     public Candidate getCandidateById(Long id) {
-        return null;
```

Comment:
always validate any input from the user, validate that the id is not null in this case